ui/src/state.ts

```diff
@@ -0,0 +1,29 @@
+export class State {
```

Comment:
What is the scope of this? Can it be named more specifically?

Reply:
It's the state for the entire app. I've renamed it Dataset instead which is more descriptive.